gpiozero/output_devices.py

```diff
@@ -898,3 +898,313 @@ def stop(self):
+        self._dc_range = (max_pulse_width - min_pulse_width) / frame_width
+        self._min_value = -1
+        self._value_range = 2
+        super(Servo, self).__init__(
+            pwm_device=PWMOutputDevice(pin, frequency=int(1 / frame_width)))
+        try:
+            self.value = initial_value
+        except:
+            self.close()
+            raise
+
+    @property
+    def frame_width(self):
+        """
+        The time between control pulses, measured in seconds.
+        """
+        return self._frame_width
+
+    @property
+    def min_pulse_width(self):
+        """
+        The control pulse width corresponding to the servo's minimum position,
+        measured in seconds.
+        """
+        return self._min_dc * self.frame_width
+
+    @property
+    def max_pulse_width(self):
+        """
+        The control pulse width corresponding to the servo's maximum position,
+        measured in seconds.
+        """
+        return (self._dc_range * self.frame_width) + self.min_pulse_width
+
+    @property
+    def pulse_width(self):
+        """
+        Returns the current pulse width controlling the servo.
+        """
+        if self.pwm_device.pin.frequency is None:
+            return None
+        else:
+            return self.pwm_device.pin.state * self.frame_width
+
+    def min(self):
+        """
+        Set the servo to its minimum position.
+        """
+        self.value = -1
+
+    def mid(self):
+        """
+        Set the servo to its mid-point position.
+        """
+        self.value = 0
+
+    def max(self):
+        """
+        Set the servo to its maximum position.
+        """
+        self.value = 1
+
+    def detach(self):
+        """
+        Temporarily disable control of the servo. This is equivalent to
+        setting :attr:`value` to ``None``.
+        """
+        self.value = None
+
+    def _get_value(self):
+        if self.pwm_device.pin.frequency is None:
+            return None
+        else:
+            return (
+                ((self.pwm_device.pin.state - self._min_dc) / self._dc_range) *
+                self._value_range + self._min_value)
+
+    @property
+    def value(self):
+        """
+        Represents the position of the servo as a value between -1 (the minimum
+        position) and +1 (the maximum position). This can also be the special
+        value ``None`` indicating that the servo is currently "uncontrolled",
+        i.e. that no control signal is being sent. Typically this means the
+        servo's position remains unchanged, but that it can be moved by hand.
+        """
+        result = self._get_value()
+        if result is None:
+            return result
+        else:
+            # NOTE: This round() only exists to ensure we don't confuse people
+            # by returning 2.220446049250313e-16 as the default initial value
+            # instead of 0. The reason _get_value and _set_value are split
```

Comment:
Ahh ... there was but it wound up being redundant so I reverted that ... obviously forgot to revert that comment though! Oh well.